MapboxCoreNavigation/RouteController.swift

```diff
@@ -744,11 +744,26 @@ extension RouteController: CLLocationManagerDelegate {
             if let error = error {
                 return completion(nil, error)
             }
-            guard let route = routes?.first else {
+            
+            guard let routes = routes else {
                 return completion(nil, nil)
             }
 
-            return completion(route, error)
+            if let route = self?.selectRouteBySimilarity(from: routes) {
+                return completion(route, error)
+            } else if let route = routes.first {
+                return completion(route, error)
+            } else {
+                return completion(nil, nil)
+            }
+        }
+    }
+    
+    func selectRouteBySimilarity(from routes: [Route]) -> Route? {
```

Comment:
This method returns a value without any side effects, so start the name with a noun, not a verb: `mostSimilarRoute(in:)`.